trace-reader/src/processing.rs

```diff
@@ -0,0 +1,144 @@
+//! This module allows one to simulate instances of DigitizerAnalogTraceMessage
+//! using the FlatBufferBuilder.
+
+use super::loader::{TraceFile, TraceFileEvent};
+use anyhow::{Error, Result};
+use chrono::Utc;
+
+use rand::Rng;
+use rdkafka::{
+    producer::{FutureProducer, FutureRecord},
+    util::Timeout,
+};
+use std::{ops::RangeInclusive, time::Duration};
+
+use common::{Channel, DigitizerId, FrameNumber, Intensity};
+use streaming_types::{
+    dat1_digitizer_analog_trace_v1_generated::{
+        finish_digitizer_analog_trace_message_buffer, ChannelTrace, ChannelTraceArgs,
+        DigitizerAnalogTraceMessage, DigitizerAnalogTraceMessageArgs,
+    },
+    flatbuffers::{FlatBufferBuilder, WIPOffset},
+    frame_metadata_v1_generated::{FrameMetadataV1, FrameMetadataV1Args, GpsTime},
+};
+
+/// Reads the contents of trace_file and dispatches messages to the given Kafka topic.
+pub(crate) async fn dispatch_trace_file(
+    mut trace_file: TraceFile,
+    trace_event_indices: Vec<usize>,
+    producer: &FutureProducer,
+    topic: &str,
+    timeout_ms: u64,
+) -> Result<()> {
+    let mut fbb = FlatBufferBuilder::new();
+    for index in trace_event_indices {
+        let event = trace_file.get_trace_event(index)?;
+        create_partly_random_message(
+            &mut fbb,
+            Utc::now().into(),
+            1..=10,
+            1..=10,
+            trace_file.get_num_channels(),
+            trace_file.get_num_samples(),
+            &event,
+        )?;
+
+        let future_record = FutureRecord::to(topic).payload(fbb.finished_data()).key("");
+        let timeout = Timeout::After(Duration::from_millis(timeout_ms));
+        match producer.send(future_record, timeout).await {
+            Ok(r) => log::debug!("Delivery: {:?}", r),
```

Comment:
Prefer adding `use log::{debug, error};` to the module imports over qualifying with `log::`.